payment/app.py

```diff
@@ -22,29 +24,64 @@ def close_db_connection():
 
 @app.post('/create_user')
 def create_user():
-    pass
+    user_id = db.incr('user_id')
+    user = {'user_id': user_id, 'credit': 0}
+    db.hset(f'user:{user_id}', mapping=user)
+    return user, 200
 
 
 @app.get('/find_user/<user_id>')
 def find_user(user_id: str):
-    pass
+    user = db.hmget(f'item:{user_id}', 'credit')
+    if None in user:
+        return {"Error": "User not found"}, 404
+
+    return {'user_id': user_id, 'credit': int(user[0])}, 200
 
 
 @app.post('/add_funds/<user_id>/<amount>')
 def add_credit(user_id: str, amount: int):
-    pass
+    db.hincrby(f'item:{user_id}', 'credit', int(amount))
+    return {'done': True}, 200
 
 
 @app.post('/pay/<user_id>/<order_id>/<amount>')
 def remove_credit(user_id: str, order_id: str, amount: int):
-    pass
+    credit = db.hget(f'item:{user_id}', 'credit')
+    if int(credit) < int(amount):
+        return {"Error:", "Not enough credit"}, 400
+    p = db.pipeline(transaction=True)
+    p.hincrby(f'user:{user_id}', 'credit', -int(amount))
+    p.hset(f'order:{order_id}', 'status', 'paid')
```

Comment:
This line should not be done in the payment service, as it makes a request to the order db. I think that we can jugde on the response code whether the payment succeeded (and should be updated in the other db)

Reply:
Does it make a request to the order db?

Reply:
@pascalbenschopTU The way I meant is was like:

- Order service calls this endpoint
- This returns either 4xx or 2xx
- If it returns 4xx the payment failed so the status is still unpaind, otherwise update to paid